`Eq(x, oo)` remains unevaluated

arihantparsoya opened the pull request:
Fixes #10927 and  #12501.

#### Output before this PR:
```python
>>> from sympy import Eq, oo
>>> from sympy.abc import x
>>> Eq(x, oo)
False
```

#### Output after this PR:
```python 
>>> from sympy import Eq, oo
>>> from sympy.abc import x
>>> Eq(x, oo)
Eq(x, oo)
```

<!-- Please give this pull request a descriptive title. Pull requests with descriptive titles are more likely to receive reviews. Describe what you changed! A title that only references an issue number is not descriptive. -->

<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below. -->

arihantparsoya committed:
Fixed Eq(x, oo)

Fixes #10927.

Changed condition from `any()` to `all()` since the equations should be
unevaluated if it contains non-valued symbols.

arihantparsoya committed:
Added tests for pretty print

Fixes #12501.

smichr commented on sympy/core/relational.py:
```diff
@@ -345,7 +345,7 @@ def __new__(cls, lhs, rhs=0, **options):
                                 rv = fuzzy_bool(Eq(*args))
                                 if rv is True:
                                     rv = None
-                elif any(a.is_infinite for a in Add.make_args(n)):  # (inf or nan)/x != 0
```
I think this is not the right place to catch your condition. The intent here is to return False for `Eq(foo-oo,0)` since there is no `foo` that can make the lhs 0. It's better to do your test with the arguments, perhaps up where `fin` is defined: 

```
if None not in fin: 
    if L != R: 
        return S.false 
    if L is False: 
        return S.true 
elif False in fin and None in fin:  # something like this...
    return None
```

Or maybe that test needs to happen if there is no denominator...Not sure and don't have time to think this out right now.

parsoyaarihant replied:
Thanks, I will work it out.

arihantparsoya committed:
added conditional in fin

Based on the discussion in github, I have removed the conditional from
line 350 since it was meant to handle expressions such as `Eq(foo-oo,
0)`.

The condition checks if there is unknown variable and a infinite
variable and returns the class object if there is.

parsoyaarihant commented on sympy/core/relational.py:
```diff
@@ -316,6 +316,8 @@ def __new__(cls, lhs, rhs=0, **options):
                 if L is False:
                     return S.true
+            elif None in fin and False in fin:
+                return Relational.__new__(cls, lhs, rhs, **options)
```
@smichr , is this fine?

smichr commented:
Looks good and tests pass. It's in.

isuruf commented:
Does this also fix `Eq(-oo, oo) == True`? If not, I'll open another PR

jksuom commented:
It seems that there is still some room for improvement:
```
>>> Eq(oo, oo)
True
>>> Eq(oo, -oo)
True
```

parsoyaarihant commented:
@jksuom , @isuruf I have addressed `Eq(oo, -oo)` issue in #12577.
